modules/project-comments/server/routes/comment.routes.js

```diff
@@ -58,7 +58,7 @@ module.exports = function (app) {
     .get (routes.setAndRun (CommentModel, function (model, req) {
       return model.getCommentsForTarget (req.params.targettype, req.params.targetid, req.params.type);
     }));
-  app.route ('/api/comments/period/:periodId/all').all(policy ('guest'))
+  app.route ('/api/comments/period/:periodId/all').all(policy ('user'))
```

Comment:
This was exposing unpublished comments to the front end, where it was then being removed. (fixed).